### PR TITLE
Upgrade to drone v0.5

### PR DESCRIPTION
**Pull request by olafurpg**
The template is copy-pasted from:
https://scalacenter.github.io/platform-staging/ci-integration.html#creating-the-drone.yml-file

The CI now runs two parallel tasks: ci-fast and ci-slow. The fast task runs unit tests and doc generation. For most PRs, a green response from this task is enough to merge. The slow tests run large property tests on the parser/tokenizer as well as scripted sbt plugin tests.

### CODE REVIEW

**Comment by olafurpg** [comment]
CI passes, ship it! :shipit:

File: .drone.yml [hunk 1]
```diff
@@ -1,20 +1,25 @@
...
+    restore: true
+    mount:
+      - /drone/.ivy2
+      - /drone/.coursier-cache
+      - /drone/.sbt
+      - /drone/.git
+
   tests:
-    image: ensime/ensime:v2.x-cache
+    image: scalacenter/scala-extras:1.0
```
Comment (xeno-by) [inline .drone.yml]: What does `extras` mean?

File: .drone.yml [hunk 2]
```diff
@@ -1,20 +1,25 @@
...
     pull: true
-    environment:
-      - COURSIER_CACHE=/drone/cache/coursier
     commands:
-      - git log | head -n 20
+      - git fetch --tags && git log | head -n 20
```
Comment (xeno-by) [inline .drone.yml]: Why is `fetch --tags` necessary?

File: .drone.yml [hunk 3]
```diff
@@ -28,14 +33,19 @@ build:
...
-  mount:
-    - /drone/cache
-notify:
-  email:
-    from: noreply@drone.geirsson.com
-    host: smtp.mailgun.org
-    username: noreply@drone.geirsson.com
-    password: $MAILGUN_PASSWORD
-    recipients:
-      - olafurpg+scalameta-ci@gmail.com
```
Comment (xeno-by) [inline .drone.yml]: Will this functionality be resurrected?

File: project/os.scala [hunk 4]
```diff
@@ -126,7 +126,7 @@ object git {
   }
 
   def distance(from: String, to: String): Int = {
-    shell.check_output(s"git rev-list $to ^$from --count", cwd = ".").trim.toInt
+    shell.check_output(s"git rev-list $to $from --count", cwd = ".").trim.toInt
```
Comment (xeno-by) [inline project/os.scala]: Why was this necessary?

File: .drone.yml (hunk 3 above)
Comment (olafurpg) [inline .drone.yml]: Yes, this should be much easier on drone v0.5, I opened #744 to enable email notifications on build failure.

File: project/os.scala (hunk 4 above)
Comment (olafurpg) [inline project/os.scala]: Because the command kept crashing. I don't know why this behavior changed, it could be caused by the git version installed on the docker image.

File: .drone.yml (hunk 1 above)
Comment (jvican) [inline .drone.yml]: No, `scala-extras` is not the image meant to be used for the platform, we already have one for that already.
  
This has been named after the fact that it does not only bundle scala and sbt, but contains extra stuff like jekyll, git, and all sort of utils that may be handy to do scripting in sbt.

Comment (olafurpg) [inline .drone.yml]: Maybe `scalacenter/ci` is a better name then? I don't really mind either way
